### PR TITLE
#5852 move neo loader

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [ ] Closes #<issue_number>
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at d061180</samp>

Replaced the Buefy loading component with a custom one from the `@kodadot1/brick` library, using Oruga as a base. Created and exported the `NeoLoading` component, defined its styles and story, and updated the `Loader.vue` component to use it. Removed the unused styles for the `loading-overlay` class.

### <samp>🤖 Generated by Copilot at d061180</samp>

> _`NeoLoading` shines_
> _Oruga replaces Buefy_
> _Winter of changes_

### CODE REVIEW

File: libs/ui/src/components/NeoLoading/NeoLoading.vue [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <o-loading
+    :active.sync="isLoadingActive"
+    :can-cancel="canCancel"
+    :overlay-class="'neo-loading-overlay'"
+    :is-full-page="isFullPage">
+    <slot />
+  </o-loading>
+</template>
+
+<script lang="ts" setup>
```
Comment (preschian) [inline libs/ui/src/components/NeoLoading/NeoLoading.vue:11]: use vue.extends. example: https://github.com/kodadot/nft-gallery/blob/main/libs/ui/src/components/NeoCarousel/NeoCarousel.vue

**Review by preschian — CHANGES_REQUESTED** [review]

File: libs/ui/src/components/NeoLoading/NeoLoading.vue (hunk 1 above)
Comment (prachi00) [reply libs/ui/src/components/NeoLoading/NeoLoading.vue:11]: I tried doing this, did not work though, maybe model didnt sync or something?

Comment (preschian) [reply libs/ui/src/components/NeoLoading/NeoLoading.vue:11]: aahh I see, let me check

Comment (preschian) [reply libs/ui/src/components/NeoLoading/NeoLoading.vue:11]: oh, yes need to sync. change `v-model` in `Loader.vue` to `:active.sync`

ref: https://vue2.oruga.io/components/Loading.html#props

```
Whether modal is active or not, use the .sync modifier (Vue 2.x) or v-model:active (Vue 3.x) to make it two-way binding
```

File: libs/ui/src/components/NeoLoading/NeoLoading.vue [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <o-loading
+    :active.sync="isLoadingActive"
+    :can-cancel="canCancel"
+    :overlay-class="'neo-loading-overlay'"
```
Comment (roiLeo) [inline libs/ui/src/components/NeoLoading/NeoLoading.vue:5]:
```suggestion
    overlay-class="neo-loading-overlay"
```

or maybe create `has-overlay` prop with default `true`

**Review by vikiival — COMMENTED** [review]
Can you please provide screenshot?

**Comment by prachi00** [comment]
only loader changed is this one @vikiival 
Screen Shot 2023-05-15 at 8 34 20 PM

File: libs/ui/src/components/NeoLoading/NeoLoading.vue (hunk 1 above)
Comment (prachi00) [reply libs/ui/src/components/NeoLoading/NeoLoading.vue]: done the changes

**Review by roiLeo — APPROVED** [review]
✅ wfm (new overlay?)

**before**
Screenshot 2023-05-16 at 08-34-10 Blurred Doggo KodaDot - NFT Market Explorer

**after**
Screenshot 2023-05-16 at 08-33-56 2 pups KodaDot - NFT Market Explorer

**Review by preschian — APPROVED** [review]

**Comment by vikiival** [comment]
>  wfm (new overlay?)

Please @prachi00 make sure that it also works in dark mode ^-^

**Comment by prury** [comment]
lmk when i should check it

**Comment by prachi00** [comment]
updated overlay
Screen Shot 2023-05-16 at 7 49 52 PM
Screen Shot 2023-05-16 at 7 49 58 PM

**Comment by preschian** [comment]
wait, is this reverted? [`cba492c`](https://github.com/kodadot/nft-gallery/pull/5933/commits/cba492cf02a40e54a8d4664ff5c48d1db677c9e1)

**Review by roiLeo — CHANGES_REQUESTED** [review]
Capture d’écran 2023-05-17 à 8 49 40 AM

**Review by preschian — APPROVED** [review]

**Review by vikiival — APPROVED** [review]
Testing along with canary, looks same

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $20 @ 25.23 USD/KSM ~ 0.793 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0x707d62decc7c9701328f21ead538853f3ac6d4ef2233aca2ca0460cde92590a0](https://kusama.subscan.io/extrinsic/0x707d62decc7c9701328f21ead538853f3ac6d4ef2233aca2ca0460cde92590a0)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**